### PR TITLE
Speech GCS

### PR DESCRIPTION
**Pull request by gguuss**
Adds examples showing transcription from a GCS URI.

### CODE REVIEW

File: speech/cloud-client/transcribe.py [hunk 1]
```diff
@@ -57,13 +55,48 @@ def main(speech_file):
     # [END send_request]
 
 
+def transcribe_gcs(gcs_uri):
+    """Transcribes the audio file specified by the gcs_uri."""
+    # [START authenticating_gcs]
```
Comment (jonparrott) [inline speech/cloud-client/transcribe.py]: Meta: seems unnecessary to chop this sample up into so many regions.

File: speech/cloud-client/transcribe.py [hunk 2]
```diff
@@ -57,13 +55,48 @@ def main(speech_file):
...
+    # Application default credentials provided by env variable
+    # GOOGLE_APPLICATION_CREDENTIALS
+    from google.cloud import speech
+    speech_client = speech.Client()
+    # [END authenticating]
+
+    # [START construct_request_gcs]
+    audio_sample = speech_client.sample(
+        None,
```
Comment (jonparrott) [inline speech/cloud-client/transcribe.py]: What is `None` here?

File: speech/cloud-client/transcribe.py [hunk 3]
```diff
@@ -57,13 +55,48 @@ def main(speech_file):
...
+    # [END construct_request_gcs]
+
+    # [START send_request_gcs]
+    alternatives = speech_client.speech_api.sync_recognize(audio_sample)
+    for alternative in alternatives:
+        print('Transcript: {}'.format(alternative.transcript))
+    # [END send_request_gcs]
+
+
+def main(path):
```
Comment (jonparrott) [inline speech/cloud-client/transcribe.py]: We generally don't put main functions in our samples, just place this logic below or do separate subparsers for local path and gcs path.

File: speech/cloud-client/transcribe_async.py [hunk 4]
```diff
@@ -71,13 +69,62 @@ def main(speech_file):
...
+    operation = speech_client.speech_api.async_recognize(audio_sample)
+
+    retry_count = 100
+    while retry_count > 0 and not operation.complete:
+        retry_count -= 1
+        time.sleep(2)
+        operation.poll()
+
+    if not operation.complete:
+        print("Operation not complete and retry limit reached.")
```
Comment (jonparrott) [inline speech/cloud-client/transcribe_async.py]: single quotes everywhere.

File: speech/cloud-client/transcribe_async.py [hunk 5]
```diff
@@ -71,13 +69,62 @@ def main(speech_file):
...
+    # [END construct_request_gcs]
+
+    # [START send_request_gcs]
+    operation = speech_client.speech_api.async_recognize(audio_sample)
+
+    retry_count = 100
+    while retry_count > 0 and not operation.complete:
+        retry_count -= 1
+        time.sleep(2)
+        operation.poll()
```
Comment (jonparrott) [inline speech/cloud-client/transcribe_async.py]: I thought they were going to change this interface soon.... hmm. @dhermes?

File: speech/cloud-client/transcribe_async_test.py [hunk 6]
```diff
@@ -16,8 +16,15 @@
...
-def test_main(resource, capsys):
+def test_transcribe(resource, capsys):
     main(resource('audio.raw'))
     out, err = capsys.readouterr()
 
     assert re.search(r'how old is the Brooklyn Bridge', out, re.DOTALL | re.I)
+
+
+def test_transcribe_gcs(resource, capsys):
+    main('gs://cloud-samples-tests/speech/brooklyn.flac')
```
Comment (jonparrott) [inline speech/cloud-client/transcribe_async_test.py]: I prefer these to be in the python-docs-samples-tests bucket, provided by `cloud_config.bucket`.

File: speech/cloud-client/transcribe_async.py (hunk 5 above)
Comment (dhermes) [inline speech/cloud-client/transcribe_async.py]: `_OperationFuture` has landed in GAX/GAPIC interfaces and we are slowly working it in

/cc @lukesneeringer

Comment (jonparrott) [inline speech/cloud-client/transcribe_async.py]: Alright, cool, we'll catch it when @dpebot bumps our dependencies.

File: speech/cloud-client/transcribe.py (hunk 2 above)
Comment (gguuss) [inline speech/cloud-client/transcribe.py]: content as in there's either content or a gcs_uri

File: speech/cloud-client/transcribe.py (hunk 1 above)
Comment (gguuss) [inline speech/cloud-client/transcribe.py]: +1, these aren't even referenced anywhere in the docs yet, I'd just stuck to the inherited convention.

File: speech/cloud-client/transcribe.py (hunk 3 above)
Comment (gguuss) [inline speech/cloud-client/transcribe.py]: Done, moved logic to right after the args are parsed

File: speech/cloud-client/transcribe_async.py (hunk 4 above)
Comment (gguuss) [inline speech/cloud-client/transcribe_async.py]: Done, I should add a module to my pavlok for such transgressions.

File: speech/cloud-client/transcribe_async_test.py (hunk 6 above)
Comment (gguuss) [inline speech/cloud-client/transcribe_async_test.py]: Done.

File: speech/cloud-client/transcribe_async_test.py [hunk 7]
```diff
@@ -13,11 +13,18 @@
 
 import re
 
-from transcribe_async import main
+from transcribe_async import transcribe_file, transcribe_gcs
```
Comment (jonparrott) [inline speech/cloud-client/transcribe_async_test.py]: Final nit: import modules, not names. Just do `import transcribe_async` and use `transcribe_async.transcribe_file`, etc.

File: speech/cloud-client/transcribe_async.py (hunk 5 above)
Comment (gguuss) [inline speech/cloud-client/transcribe_async.py]: I'll skip this for now.

**Pull request merged**